aws/terminator/compute.py

```diff
@@ -303,7 +303,17 @@ def name(self):
     def created_time(self):
         return self.instance['createdAt']
 
+    # EKS Fargate Profile is cluster dependent and can only be listed or described through it.
+    def _find_eks_fargate_profile(self):
+        return self.client.list_fargate_profiles(clusterName=self.name)['fargateProfileNames']
+
     def terminate(self):
+        for profile in self._find_eks_fargate_profile():
+            self.client.delete_fargate_profile(clusterName=self.name, fargateProfileName=profile)
+            # To delete a cluster it cannot have any active fargates,
+            # so we need to delete it and wait for the end to proceed.
+            waiter = self.client.get_waiter('fargate_profile_deleted')
+            waiter.wait(clusterName=self.name, fargateProfileName=profile)
```

Comment:
Instead of waiting for the delete, why not separate the cluster deletion from the fargate profile deletion? We do that for other resources that have dependencies.

Reply:
I really don't know if its the best way to do it, but I implemented this way trying to solve 2 major problems:

1- As fargates are cluster dependent, we have no way to list/describe a fargate without informing a cluster it belongs to.

2- The process of deleting a cluster is also dependent on the existence or not of the fargate in that cluster, if you have one fargate created or in the state "deleting" (it takes about 2 minutes to delete a fargate) the deleting process fails. This also occurs between fargates, for example if you have two fargates and one is in the "deleting" state, the other can only be deleted if the first one ends, otherwise a failure occurs.
To solve this i added the waiter, this way we guarantee that you won't have any fargate created or being deleted avoiding the error.

Reply:
Any delete that takes 2 minutes is something we shouldn't be waiting on. That will block the Lambda function execution for too long. There must be some way we can check on the status of the delete on subsequent runs of the function, rather than waiting for it to complete.

Reply:
I can remove the waiter and add a try except statement on the deletions, that way if it fails we just escape and in the next lambda run it will remove it.
What do you think @mattclay?

Reply:
Yes, that should work, and is generally how we handle the situation with other resources with dependencies.

In most cases we have a class for each resource type to delete, with the shortest lifetimes defined for the ones that must be deleted first.

Reply:
I changed it to a dedicated Eks Fargate Profile class and added the try except statement. I think it's better now.